src/main/java/decodes/tsdb/TimeSeriesDb.java

```diff
@@ -576,64 +571,65 @@ public abstract class TimeSeriesDb
     /**
      * No args constructor required.
      */
-    public TimeSeriesDb()
+    public TimeSeriesDb(String appName, javax.sql.DataSource dataSource, DecodesSettings settings) throws DatabaseException
```

Comment:
appName:    It feels odd to require TimeSeriesDb, OpenTsDb, CwmsTimeSeriesDb, and HdbTimeSeriesDb to need to be told the name of the application connecting.   I can see that may be needed for computation processing to keep track of different processes.

Could there be a OpenDCSDataSource instead of javax.sql.DataSource has that optional appName/appID/key if required?

Reply:
It's not actually used everywhere, but it's pervasive enough we're stuck with it.

As for renaming, that doesn't make sense to me. the javax.sql.DataSource *is* the connection or connection pool. And just that. The app name really has nothing to do with it, I would make the appname part of the `OpenDcsDatabase` class and tweak how it's used. (Mostly in logging really, which is all going away, but also to lookup properties which should be at that higher level.)

Reply:
So the AppName is used by TsImport, and at least the Postgres database, but I believe HDB as well to mark were a timeseries value was brought in from.

I do think it should go away from this level, but going to take some time to properly move around.